monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,133 @@
+        
+        usses = self.test_config['usses']
+
+        all_test_payloads = []
+        
+        for uss_index, uss in enumerate(usses):
+            requested_flights = []
+            flight_track_path = Path(self.tracks_directory, self.flight_tracks[uss['allocated_flight_track_number']])
+            with open(flight_track_path) as generated_rid_state:
+                rid_state_data = json.load(generated_rid_state)
+            
+            effective_after = rid_state_data['reference_time']
+            flight_details =  rid_state_data['flight_details']
+            operator_details = rid_state_data['operator_details']
+            
+            operator_location = OperatorLocation(lat = operator_details['location']['latitude'], lng = operator_details['location']['longitude'])
+            operator_id = str(uuid.uuid4())
+
+            rid_flight_details = RIDFlightDetails(operator_id = operator_id, operator_location = operator_location, operation_description = flight_details['operation_description'] , serial_number = flight_details['serial_number'], registration_number = flight_details['registration_number'])
+
+            test_flight_details = TestFlightDetails(effective_after= effective_after,details = rid_flight_details)
+            test_flight = TestFlight(injection_id = str(uuid.uuid4()), telemetry= rid_state_data['flight_telemetry'], details_responses = test_flight_details)            
+            test_flight_deserialized = self.make_json_compatible(test_flight)
+            requested_flights.append(test_flight_deserialized)
+            test_payload = {'test_id': str(uuid.uuid4()), "requested_flights": requested_flights}        
+
+            all_test_payloads.append({'injection_url':uss['injection_url'], 'injection_payload': test_payload, 'injection_start_time_from_now_secs':uss['start_time_from_now_secs']})        
+        
+        return all_test_payloads
+
+
+class TestHarness():
+    ''' A class to submit Aircraft RID State to the USS test endpoint '''
+
+    def __init__(self, auth_spec:str, auth_url:str):
+        self.auth_spec = auth_spec
+        self.auth_url= auth_url
+        
+    def get_dss_session(self, auth_url:str, auth_spec:str):
+        ''' This method gets a DSS session using the monitoring tools that are provided in the DSS monitoring repository'''
+
+        auth_adapter = make_auth_adapter(auth_spec)
+        s = DSSTestSession(auth_url, auth_adapter)
+    
+        return s
+
+    async def submit_test(self,dss_session, injection_url,  test_payload):
+        print(f"Started: {time.strftime('%X')}")
+        print("Waiting %f seconds" % test_payload['injection_start_time_from_now_secs'])
+        await asyncio.sleep(test_payload['injection_start_time_from_now_secs'])            
+        response = dss_session.put(injection_url, data=test_payload['injection_payload'])
```

Comment:
The session here already has a base URL prefix (`injection_url`), so what we want here is rather [the specific endpoint we want to address](https://github.com/interuss/dss/blob/fd1d6f4a0a7bc07da88e3dade15e6c8672dd779a/interfaces/automated-testing/rid/injection.yaml#L9), and note that the endpoint path requires that we include a test ID.

Reply:
OK I have now fixed this, and it should work as you describe (I think).